google/cloud/spanner/connection.h

```diff
@@ -54,9 +55,27 @@ class Connection {
     KeySet keys;
     std::vector<std::string> columns;
     ReadOptions read_options;
+
+    // TODO(#307): Refactor once correct location for session implemented.
+    google::cloud::optional<std::string> session_name;
+
+    ReadParams(Transaction transaction, std::string table, KeySet keys,
```

Comment:
Do we need this constructor? It looks like all the callers are using aggregate initialization, so I'm not sure this is needed.

Reply:
As ReadParams is likely a temporary container for sessions_name, I added the constructor so that the we don't need to change the call sites where we are not specifying session_name. When we remove session_name from ReadParams, we can remove this constructor.